atc/auditor/auditor.go

```diff
@@ -0,0 +1,174 @@
+package auditor
+
+import (
+	"net/http"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/concourse/concourse/atc"
+)
+
+//go:generate counterfeiter . Auditor
+
+
+func NewAuditor(
+	EnableBuildAuditLog bool,
+	EnableContainerAuditLog bool,
+	EnableJobAuditLog bool,
+	EnablePipelineAuditLog bool,
+	EnableResourceAuditLog bool,
+	EnableSystemAuditLog bool,
+	EnableTeamAuditLog bool,
+	EnableWorkerAuditLog bool,
+	EnableVolumeAuditLog bool,
+	logger lager.Logger,
+) *auditor {
+	return &auditor{
+		EnableBuildAuditLog:     EnableBuildAuditLog ,
+		EnableContainerAuditLog: EnableContainerAuditLog,
+		EnableJobAuditLog:       EnableJobAuditLog,
+		EnablePipelineAuditLog:  EnablePipelineAuditLog,
+		EnableResourceAuditLog:  EnableResourceAuditLog,
+		EnableSystemAuditLog:    EnableSystemAuditLog,
+		EnableTeamAuditLog:      EnableTeamAuditLog,
+		EnableWorkerAuditLog:    EnableWorkerAuditLog,
+		EnableVolumeAuditLog:    EnableVolumeAuditLog,
+		logger:                  logger,
+	}
+}
+
+
+type Auditor interface {
+	Audit(action string, userName string, r *http.Request)
+}
+
+type auditor struct {
+	EnableBuildAuditLog bool
+	EnableContainerAuditLog bool
+	EnableJobAuditLog bool
+	EnablePipelineAuditLog bool
+	EnableResourceAuditLog bool
+	EnableSystemAuditLog bool
+	EnableTeamAuditLog bool
+	EnableWorkerAuditLog bool
+	EnableVolumeAuditLog bool
+	logger lager.Logger
+}
+
+func (a *auditor) ValidateAction(action string) bool {
+	switch loggingLevels[action] {
+	case "EnableBuildAuditLog":
+		return a.EnableBuildAuditLog
+	case "EnableContainerAuditLog":
+		return a.EnableContainerAuditLog
+	case "EnableJobAuditLog":
+		return a.EnableJobAuditLog
+	case "EnablePipelineAuditLog":
+		return a.EnablePipelineAuditLog
+	case "EnableResourceAuditLog":
+		return a.EnableResourceAuditLog
+	case "EnableSystemAuditLog":
+		return a.EnableSystemAuditLog
+	case "EnableTeamAuditLog":
+		return a.EnableTeamAuditLog
+	case "EnableWorkerAuditLog":
+		return a.EnableWorkerAuditLog
+	case "EnableVolumeAuditLog":
+		return a.EnableVolumeAuditLog
+	default:
+		return false
+	}
+}
+
+func (a *auditor) Audit(action string, userName string, r *http.Request) {
+		err := r.ParseForm()
+		if err == nil &&  a.ValidateAction(action) {
+			a.logger.Info("audit", lager.Data{"command": action, "user": userName, "parameters": r.Form})
```

Comment:
Would you mind calling this `"action"` instead of `"command"`? Minor nit, but worth checking early since these logs will likely be machine processed at some point.

Reply:
Absolutely, I debated back and forth as to put it as "action", I can't remember why I landed on "command", but I can change it.